File: bench/agent-eval/fixtures/ctf-permissions/app-ts/src/routes/import.ts
// Import routes. This gates a DIFFERENT concept, the IMPORT permission
// (`can_import` / PERM_IMPORT / canImport). Structurally parallel to export so
// it is an attractive false positive; it is the negative-control target (a
// task asks for import gates specifically).
import { User, currentUser } from "../models/user";
import { permissions } from "../perm/service";
import { PERM_IMPORT } from "../perm/constants";

// release 1 style: inline flag, but the IMPORT flag.
export function importCsv(req: any, res: any): void {
  const user: User = currentUser(req);
  if (!user.canImport) {
    res.status(403).send("no import");
    return;
  }
  res.send("imported");
}

// release 2 style: service, but the IMPORT permission.
export function importJson(req: any, res: any): void {
  const user: User = currentUser(req);
  permissions.require(user, PERM_IMPORT);
  res.send("imported");
}

File: v5/bench/agent-eval/fixtures/ctf-permissions/app-rs/src/handlers/import.rs
//! Import handlers. DECOY concept: these gate the IMPORT permission
//! (`can_import`), not export. Negative-control target.
use crate::models::{PermResult, User};
use crate::perm_flags::check_import_flag;
use crate::perm_guard::require_import;

// release 1 style, import concept.
pub fn import_csv(user: &User) -> PermResult {
    check_import_flag(user)?;
    Ok(())
}

// release 3 style, import concept.
pub fn import_bulk(user: &User) -> PermResult {
    require_import(user)?;
    Ok(())
}
